### PR TITLE
Use (<>) from Data.Monoid

### PR DESCRIPTION
**Pull request by saep**
I was a bit annoyed to always qualify or hide (<>) in the import list.

### CODE REVIEW

**Comment by hvr** [comment]
@saep you need to take into account that `<>` exists only since `base-4.5` in `Data.Monoid`

**Comment by saep** [comment]
I'm actually not sure whether I should `#ifdef` the code with base/ghc checks or just accept that this would need a ghc of 7.4 or newer.

**Comment by ekmett** [comment]
I tend to #ifdef off of `__GLASGOW_HASKELL__` for base bounds:

@hvr maintains a [version history](https://ghc.haskell.org/trac/ghc/wiki/Commentary/Libraries/VersionHistory?action=history) that provides a nice mapping back and forth between base version and GHC version.

Note: This only really works for this package. Every other pretty printer has a slightly different associativity for `(<>)` -- and this can affect performance exponentially for them.

**Comment by saep** [comment]
Alright, here's an `#ifdeffed` version. (The example compiles, runs and looks correct on GHC 7.10.2 and 7.6.3.)

File: Text/PrettyPrint/ANSI/Leijen.hs [hunk 1]
```diff
@@ -151,13 +151,16 @@ import System.Console.ANSI (Color(..), ColorIntensity(..), ConsoleLayer(..),
 
 import Data.String (IsString(..))
 import Data.Maybe (catMaybes)
-#if __GLASGOW_HASKELL__ < 710
-import Data.Monoid (Monoid, mappend, mconcat, mempty)
+#if __GLASGOW_HASKELL__ >= 710
+import Data.Monoid ((<>))
+#elif __GLASGOW_HASKELL__ >= 704
+import Data.Monoid (Monoid, mappend, mconcat, mempty, (<>))
```
Comment (hvr) [inline Text/PrettyPrint/ANSI/Leijen.hs:157]: This will fail on GHC 7.0/7.2

**Comment by hvr** [comment]
@ekmett what kind of version bump are you considering? is it ok if we drop support for GHC < 7.4?

**Comment by ekmett** [comment]
I'd prefer to keep GHC < 7.4. This is a very broadly used package.

**Comment by hvr** [comment]
@ekmett well, can you enable travis-ci for this repo?

**Comment by ekmett** [comment]
Sure.

**Comment by ekmett** [comment]
Even better: I added you as a collaborator, so you can do it. ;)

**Comment by hvr** [comment]
@ekmett I'm afraid I don't seem to be able to enable Travis myself for this repo

**Comment by ekmett** [comment]
I've turned the feature on, but it now needs one of your magical multi-ghc travis configs. ;)

**Pull request merged**